### PR TITLE
Forward list

### LINKED ISSUE #5317: [Concept Entry] C++ Forward List forward_list
### Reason/inspiration (optional)

We would like a new entry on the `Forward_List` in C++. The entry should go in a new file under `docs/content/cpp/concepts/forward-list/forward-list.md`.       
  
The entry should include:      
  
- An introduction to the concept      
- A `Syntax` section that provides the syntax for the concept (if applicable)      
- An `Example` section that provides an example demonstrating the concept in use (if applicable)     
 - An `Codebyte Example` section that provides an example demonstrating how to use deque
 
Please refer to the [concept entry template](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.|

### Entry Type (select all that apply)

New Concept Entrty for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by AlejandroAndrouin**
### Issue Solved

Closes #5317 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Comment by cigar-galaxy82** [comment]
I think we can remove the Useful function section for now for this PR and keep everything before it in the document and as the whole thing is very large for docs standard

**Review by cigar-galaxy82 — CHANGES_REQUESTED** [review]

**Comment by cigar-galaxy82** [comment]
Hi @AlejandroAndrouin thank you for contributing to Codecademy docs, I have suggested a few changes please modify them.

File: content/cpp/concepts/forward-list/forward-list.md [hunk 1]
```diff
@@ -0,0 +1,248 @@
+---
+Title: 'Forward List'
+Description: 'Forward List is a sequence container that allows efficient insertions and deletions from the front of the list.'
```
Comment (Radhika-okhade) [inline content/cpp/concepts/forward-list/forward-list.md:3]: Description should always start with a verb.

File: content/cpp/concepts/forward-list/forward-list.md [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+---
+
+A **`Forward List`** is a sequence container that allows efficient insertions and deletions from the front of the list. Forward lists work by storing a memory address to the first element of the list and having each element store data and a pointer to the memory location of the next element. Compared to arrays, forward lists offer quicker insertions and deletions at the start of the list, at the cost of direct access to elements by index.
+
+## Syntax
+
+```pseudo
+#include <forward_list>
+
+std::forward_list<type> forwardlist;
```
Comment (Radhika-okhade) [inline content/cpp/concepts/forward-list/forward-list.md:22]:
```suggestion
std::forward_list<type> forwardlist; // Replace 'type' with any data type as needed.
```

File: content/cpp/concepts/forward-list/forward-list.md [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+  std::cout << element << " ";
+}
+std::cout << std::endl;
+```
+
+```shell
+Elements: 1 2 3 4 5
+Calling list.assign({6, 7})
+Elements: 6 7
+```
```
Comment (Radhika-okhade) [inline content/cpp/concepts/forward-list/forward-list.md:248]: Please remove all this and just a Codebyte Example section that provides an example demonstrating how to use forward_list

**Comment by Radhika-okhade** [comment]
Thank you for contributing to Codecademy Docs @AlejandroAndrouin 🙂
There are a few changes I have suggested. Could you please go through them?

**Review by Radhika-okhade — APPROVED** [review]

**Review by mamtawardhani — APPROVED** [review]
Thank you for contributing to Codecademy docs @alejandroAndrowin 😄 

The entry looks good to be merged! 🚀

**Pull request merged**